src/include/catalog/postgres/pg_statistic.h

```diff
@@ -30,13 +36,44 @@ class PgStatistic {
    */
   static constexpr CatalogColumnDef<table_oid_t, uint32_t> STARELID{col_oid_t{1}};
   static constexpr CatalogColumnDef<col_oid_t, uint32_t> STAATTNUM{col_oid_t{2}};
-  static constexpr CatalogColumnDef<uint32_t, uint32_t> STA_NULLROWS{col_oid_t{3}};
-  static constexpr CatalogColumnDef<uint32_t, uint32_t> STA_NUMROWS{col_oid_t{4}};
+  static constexpr CatalogColumnDef<uint32_t, uint32_t> STA_NUMROWS{col_oid_t{3}};
+  static constexpr CatalogColumnDef<uint32_t, uint32_t> STA_NONNULLROWS{col_oid_t{4}};
+  static constexpr CatalogColumnDef<uint32_t, uint32_t> STA_DISTINCTROWS{col_oid_t{5}};
+  static constexpr CatalogColumnDef<storage::VarlenEntry> STA_TOPK{col_oid_t{6}};
+  static constexpr CatalogColumnDef<storage::VarlenEntry> STA_HISTOGRAM{col_oid_t{7}};
 
-  static constexpr uint8_t NUM_PG_STATISTIC_COLS = 4;
+  static constexpr uint8_t NUM_PG_STATISTIC_COLS = 7;
 
   static constexpr std::array<col_oid_t, NUM_PG_STATISTIC_COLS> PG_STATISTIC_ALL_COL_OIDS = {
-      STARELID.oid_, STAATTNUM.oid_, STA_NULLROWS.oid_, STA_NUMROWS.oid_};
+      STARELID.oid_,         STAATTNUM.oid_, STA_NUMROWS.oid_,  STA_NONNULLROWS.oid_,
+      STA_DISTINCTROWS.oid_, STA_TOPK.oid_,  STA_HISTOGRAM.oid_};
+
+ public:
+  /**
+   * Contains information on how to derive values for the columns within pg_statistic
+   */
+  struct PgStatisticColInfo {
+    /** The type of aggregate to use */
+    parser::ExpressionType aggregate_type_;
+    /** Whether the aggregate is distinct */
+    bool distinct_;
+    /** Oid of the column */
+    catalog::col_oid_t column_oid_;
+    /** Human readable name of the column */
+    std::string_view name_;
```

Comment:
Yes it's just for tpl variable names to make them slightly more readable in case anyone was reading/debuging the TPL. I think using the column OIDs is just as good though.

Reply:
Updated